lib/awful/autofocus.lua

```diff
@@ -21,7 +21,7 @@ end
 --
 -- @param obj An object that should have a .screen property.
 local function check_focus(obj)
-    if not obj.screen.valid then return end
+    if (not obj.screen) or not obj.screen.valid then return end
```

Comment:
Why do you have to add "screen might be `nil`"-checks to objects now? I thought (and read) that running without any screens is still not supported, so I would expect that no new "object suddenly has no screen"-cases can appear.

Reply:
To allow the default rc.lua to work without screens. That's purely to make the CI pass. Later commits make up for the lack of screens by adding them at the right point in time to restore "sanity". However to have smaller commits that individually pass the tests, we have this commit...